### PR TITLE
[fix]: [PLG-2939]: stop credits provisioning in the wrong cluster

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes
stop credits provisioning in the wrong cluster

Local testing results
**********************
Case 1: when account does not exist in the cluster

API response: 
{
    "metaData": null,
    "resource": null,
    "responseMessages": [
        {
            "code": "ACCOUNT_DOES_NOT_EXIST",
            "level": "ERROR",
            "message": "Account does not exist.",
            "exception": null,
            "failureTypes": []
        }
    ]
}

Logs:
2023-10-21 03:10:55,649 [dw-287 - GET /api/account/kmpySmUISimoRrJL6NL73w/ng/credit?routingId=kmpySmUISimoRrJL6NL73w&clientAccountId=xyz] ERROR software.wings.resources.AccountResource - Account is not found for the given id:xyz [X-requestId=59c45e8e-0d03-4b34-bd9a-63c88a30f2f7] 
software.wings.exception.AccountNotFoundException: Account is not found for the given id:xyz
	at software.wings.service.impl.AccountServiceImpl.get(AccountServiceImpl.java:587)
	at ru.vyarus.guice.validator.aop.ValidationMethodInterceptor.invoke(ValidationMethodInterceptor.java:52)
	at software.wings.resources.AccountResource.validateAccountExistence(AccountResource.java:854)

*****************
Case 2: when account exists in the cluster but accountId mismatches with CreditDTO

API response:
API response:
{
    "metaData": null,
    "resource": null,
    "responseMessages": [
        {
            "code": "INVALID_FORMAT",
            "level": "ERROR",
            "message": "Invalid format of YAML payload: HTTP Error Status (400 - Invalid Format) received. Invalid request: AccountIdentifier: [kmpySmUISimoRrJL6NL73w] did not match with the Credit information",
            "exception": null,
            "failureTypes": []
        }
    ]
}

Logs:
2023-10-21 03:17:13,703 [dw-225 - POST /admin/credits/create?accountIdentifier=kmpySmUISimoRrJL6NL73w] WARN  io.harness.ng.core.exceptionmappers.WingsExceptionMapperV2 - Response message: Invalid request: AccountIdentifier: [kmpySmUISimoRrJL6NL73w] did not match with the Credit information
Exception occurred: AccountIdentifier: [kmpySmUISimoRrJL6NL73w] did not match with the Credit information [X-requestId=2eb02819-6b27-4a40-af0a-5fd1bbd2af17] 
io.harness.exception.InvalidRequestException: AccountIdentifier: [kmpySmUISimoRrJL6NL73w] did not match with the Credit information
	at io.harness.credit.services.impl.CreditServiceImpl.purchaseCredit(CreditServiceImpl.java:80)
	at io.harness.credit.resource.AdminCreditResource.createCredits(AdminCreditResource.java:74)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)

*****************

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54447)

### CODE REVIEW

**Comment by kapil-harness** [comment]
trigger uts

**Review by Xin-Harness — DISMISSED** [review]

File: 400-rest/src/main/java/software/wings/resources/AccountResource.java [hunk 1]
```diff
@@ -849,4 +851,13 @@ public RestResponse<Boolean> scheduleAccountLevelJobs(@PathParam("accountId") St
           .build();
     }
   }
+
+  private void validateAccountExistence(String accountIdentifier) {
+    try {
+      accountService.get(accountIdentifier);
+    } catch (Exception ex) {
+      log.error(ex.getMessage());
```
Comment (Xin-Harness) [inline 400-rest/src/main/java/software/wings/resources/AccountResource.java:859]: I recommend use a more descriptive log error message

Comment (neelam-harness) [reply 400-rest/src/main/java/software/wings/resources/AccountResource.java:859]: Log the exception as well.. 
`log.error("Account {} does not exist", accountIdentifier, ex);`

File: 930-ng-license-manager/src/main/java/io/harness/credit/services/impl/CreditServiceImpl.java [hunk 2]
```diff
@@ -71,6 +74,12 @@ public List<CreditDTO> getCredits(String accountIdentifier, CreditType creditTyp
 
   @Override
   public CreditDTO purchaseCredit(String accountIdentifier, CreditDTO creditDTO) {
+    if (!accountIdentifier.equals(creditDTO.getAccountIdentifier())) {
+      String errorMessage = "accountIdentifier did not match";
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/credit/services/impl/CreditServiceImpl.java:78]: Nit: more descriptive here too.. 
`AccountIdentifier = [%s] did not match with the Credit information`

**Review by neelam-harness — COMMENTED** [review]
LGTM overall

File: 400-rest/src/main/java/software/wings/resources/AccountResource.java (hunk 1 above)
Comment (kapil-harness) [reply 400-rest/src/main/java/software/wings/resources/AccountResource.java]: actual exception is thrown inside accountService.get(), PFB, here we are just capturing it, logging it and throwing it.
For actual logs and API response pls see the PR description.
```
throw new AccountNotFoundException(
          "Account is not found for the given id:" + accountId, null, ACCOUNT_DOES_NOT_EXIST, Level.ERROR, USER, null);
```

File: 930-ng-license-manager/src/main/java/io/harness/credit/services/impl/CreditServiceImpl.java (hunk 2 above)
Comment (kapil-harness) [reply 930-ng-license-manager/src/main/java/io/harness/credit/services/impl/CreditServiceImpl.java]: done

**Comment by kapil-harness** [comment]
trigger uts

**Review by neelam-harness — APPROVED** [review]
LGTM

**Comment by kapil-harness** [comment]
trigger message check

**Comment by kapil-harness** [comment]
trigger messagecheck

**Review by Xin-Harness — APPROVED** [review]

**Review by mohitg0795 — APPROVED** [review]

**Comment by kapil-harness** [comment]
trigger review

**Pull request merged**